guides/micronaut-oracle-cloud-streaming/chess-game/kotlin/src/main/kotlin/example/micronaut/chess/GameReporter.kt

```diff
@@ -0,0 +1,32 @@
+package example.micronaut.chess
+
+import example.micronaut.chess.dto.GameDTO
+import example.micronaut.chess.dto.GameStateDTO
+import io.micronaut.configuration.kafka.annotation.KafkaClient
+import io.micronaut.configuration.kafka.annotation.KafkaKey
+import io.micronaut.configuration.kafka.annotation.Topic
+import io.reactivex.Single
+
+@KafkaClient // <1>
+interface GameReporter {
+
+    /**
+     * Send a game indicating the start or finish (checkmate or draw).
+     *
+     * @param gameId the game id
+     * @param game the game DTO
+     * @return a reactive Single to trigger non-blocking send
+     */
+    @Topic("chessGame") // <2>
+    fun game(@KafkaKey gameId: String, game: GameDTO): Single<GameDTO> // <3> <4>
```

Comment:
Instead of 

```
    fun game(@KafkaKey gameId: String, game: GameDTO): Single<GameDTO> 
```

use: 

```
@SingleResult
    fun game(@KafkaKey gameId: String, game: GameDTO): Publisher<GameDTO> 
```